spring/spring5/spring5-common/src/main/java/org/infinispan/spring/common/session/AbstractInfinispanSessionRepository.java

```diff
@@ -75,7 +79,11 @@ public MapSession findById(String sessionId) {
 
    @Override
    public void deleteById(String sessionId) {
-      MapSession mapSession = (MapSession) cache.get(sessionId).get();
+      ValueWrapper valueWrapper = cache.get(sessionId);
+      if (valueWrapper == null) {
+         return;
+      }
+      MapSession mapSession = (MapSession) valueWrapper.get();
       if (mapSession != null) {
          applicationEventPublisher.emitSessionDeletedEvent(mapSession);
```

Comment:
I'm not sure we need to send a session deleted event when the id has changed. I need to double check this behavior with spring session people.

Reply:
Grabbing a look to other code I know that we should not call deleteById but remove the entry from the repository using Infinispan remove and skip notifications. I will add a commit with the correction soon